liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,404 @@
+      apiResults.accounts.forEach((currentAccount) => {
+        foundAccounts.push(currentAccount);
+      });
+    }));
+    // #endregion
+
+    // #region Parse Compound data into new objects
+    // Get a full list of token addresses
+    const foundTokens = foundAccounts
+      .map((currentAccount) => currentAccount.tokens.map((token) => token.address)).flat();
+
+    // Use Array to Set to Array to remove duplicates
+    const filteredTokens = Array.from(new Set(foundTokens));
+
+    // Initialize token objects first
+    await Promise.all(filteredTokens.map(async (token) => {
+      await verifyToken(data, token);
+    }));
+
+    // Loop through found accounts
+    foundAccounts.forEach((account) => {
+      // add to tracked accounts
+      /* eslint-disable no-param-reassign */
+      if (data.accounts[account.address] === undefined) data.accounts[account.address] = {};
+
+      // Add found health
+      data.accounts[account.address].health = BigNumber(account.health.value);
+      // Loop through tokens and update balances.
+      account.tokens.forEach((token) => {
+        // Process borrows as 'token'
+        if (
+          token.borrow_balance_underlying !== undefined
+          && token.borrow_balance_underlying.value !== 0
+        ) {
+          data.borrow[token.address][account.address] = BigNumber(
+            token.borrow_balance_underlying.value,
+          );
+        }
+
+        // Process supplies as 'cTokens'
+        if (
+          token.supply_balance_underlying !== undefined
+          && token.supply_balance_underlying.value !== 0
+        ) {
+          data.supply[token.address][account.address] = BigNumber(
+            token.supply_balance_underlying.value,
+          ).dividedBy(data.tokens[token.address].exchangeRateMult);
+        }
+      }); // end token loop
+      /* eslint-enable no-param-reassign */
+    }); // end account loop
+    // #endregion
+  };
+}
+
+function provideHandleTransaction(data) {
+  return async function handleTransaction(txEvent) {
+    // #region Filter logs and look for new account activity.
+    // Filter all new transactions and look for new accounts to track.
+    const borrowString = 'event Borrow(address borrower, uint256 borrowAmount, uint256 accountBorrows, uint256 totalBorrows)';
+    const exitMarketString = 'event MarketExited(address cToken, address account)';
+
+    const exitMarketEvents = txEvent.filterLog(exitMarketString, data.comptrollerAddress);
+    exitMarketEvents.forEach((exitEvent) => { data.newAccounts.push(exitEvent.args.account); });
+    const borrowEvents = txEvent.filterLog(borrowString);
+    borrowEvents.forEach((borrowEvent) => { data.newAccounts.push(borrowEvent.args.borrower); });
+    // #endregion
+
+    // Return zero findings
+    return [];
+  };
+}
+
+function provideHandleBlock(data) {
+  // eslint-disable-next-line no-unused-vars
+  return async function handleBlock(blockEvent) {
```

Comment:
```suggestion
  return async function handleBlock() {
```

Reply:
Correct, I do not use the blockEvent data, I wasn't sure if it would cause an error if I removed the parameter here.